plugins/Unity/Launcher/appdrawermodel.cpp

```diff
@@ -82,11 +84,12 @@ void AppDrawerModel::appAdded(const QString &appId)
     endInsertRows();
 }
 
-void AppDrawerModel::appRemoved(const QString &appId)
+// ubuntuAppId is appID without version suffix, see xdgwatcher.cpp.
+void AppDrawerModel::appRemoved(const QString &appId, const QString &ubuntuAppId)
 {
     int idx = -1;
     for (int i = 0; i < m_list.count(); i++) {
-        if (m_list.at(i)->appId() == appId) {
+        if (m_list.at(i)->appId() == appId || m_list.at(i)->appId() == ubuntuAppId) {
```

Comment:
Do we even need the second comparison here (and thus the argument or need to strip the version) at all? Shouldn't the `appId` that gets passed in already have the version stripped always? In what cases would it not be?

Reply:
The reason for this is to make sure we don't strip an "standard" appID by accident. Example is kate_kde.desktop, we would strip "_kde" in that case. Since the file has been removed we have no way of knowing if it's a "ubuntu style" app id.

Reply:
OK, I re-read the comment in XdgWatcher, and this is a hack to deal with the fact the file was removed. I will have to think a bit on this, but I'm certain we can handle it a better way.